samples/algorithms/SuperdenseCoding.qs

```diff
@@ -9,37 +9,39 @@
 /// This Q# program implements superdense coding to send two classical bits of
 /// information.
 namespace Sample {
+    open Microsoft.Quantum.Measurement;
     open Microsoft.Quantum.Diagnostics;
     open Microsoft.Quantum.Random;
 
     @EntryPoint()
-    operation Main() : (Bool, Bool) {
+    operation Main() : ((Bool, Bool), (Bool, Bool)) {
         use (aliceQubit, bobQubit) = (Qubit(), Qubit());
 
         // The protocol starts with the preparation of an entangled state, which
         // is later shared between Alice and Bob.
         CreateEntangledPair(aliceQubit, bobQubit);
 
         // Alice encodes 2 random bits in her qubit of the entangled pair.
-        let randomInt = DrawRandomInt(0, 3);
-        let bit1 = (randomInt &&& 1) == 0b01;
-        let bit2 = (randomInt &&& 2) == 0b10;
-        SuperdenseEncode(bit1, bit2, aliceQubit);
+        let sourceBit1 = DrawRandomBit();
+        let sourceBit2 = DrawRandomBit();
+        SuperdenseEncode(sourceBit1, sourceBit2, aliceQubit);
 
         // "Send" Alice's qubit to Bob and let Bob decode two bits.
         let (decodedBit1, decodedBit2) = SuperdenseDecode(aliceQubit, bobQubit);
 
         ResetAll([aliceQubit, bobQubit]);
-        return (decodedBit1, decodedBit2);
+        return ((sourceBit1, sourceBit2), (decodedBit1, decodedBit2));
+    }
+
+    // Generates random bit using an auxilliary qubit
+    operation DrawRandomBit() : Bool {
+        use q = Qubit();
+        H(q);
+        MResetZ(q) == One
```

Comment:
```suggestion
        return MResetZ(q) == One;
```

Reply:
nit: our samples do not generally use implicit returns (at least currently).

Reply:
aww, I liked having the implicit return in there. But yes, you are right that it's not what we've done so far. When would be the right time to start using that language feature?

Reply:
Added return.